Compilation failure on mips64el with GCC 14

Hi,

Version 0.3.28 fails to compile on mips64el (official Debian package, 32-bit indexing + pthread flavour).

Relevant part:
```
cc -c -Wdate-time -D_FORTIFY_SOURCE=2 -g -O2 -Werror=implicit-function-declaration -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security  -DMAX_STACK_ALLOC=2048 -Wall -mabi=64 -DNO_MSA -DF_INTERFACE_GFORT -fPIC -DNO_LAPACKE -DSMP_SERVER -DNO_WARMUP -DMAX_CPU_NUMBER=128 -DMAX_PARALLEL_NUMBER=1 -DUSE_SIMPLE_THREADED_LEVEL3 -DBUILD_SINGLE=1 -DBUILD_DOUBLE=1 -DBUILD_COMPLEX=1 -DBUILD_COMPLEX16=1 -DVERSION=\"0.3.28\" -UASMNAME -UASMFNAME -UNAME -UCNAME -UCHAR_NAME -UCHAR_CNAME -DASMNAME=spotrf_U_parallel -DASMFNAME=spotrf_U_parallel_ -DNAME=spotrf_U_parallel_ -DCNAME=spotrf_U_parallel -DCHAR_NAME=\"spotrf_U_parallel_\" -DCHAR_CNAME=\"spotrf_U_parallel\" -DNO_AFFINITY -I../.. -UDOUBLE  -UCOMPLEX -UCOMPLEX -UDOUBLE potrf_U_parallel.c -o spotrf_U_parallel.o
cc -c -Wdate-time -D_FORTIFY_SOURCE=2 -g -O2 -Werror=implicit-function-declaration -ffile-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security  -DMAX_STACK_ALLOC=2048 -Wall -mabi=64 -DNO_MSA -DF_INTERFACE_GFORT -fPIC -DNO_LAPACKE -DSMP_SERVER -DNO_WARMUP -DMAX_CPU_NUMBER=128 -DMAX_PARALLEL_NUMBER=1 -DUSE_SIMPLE_THREADED_LEVEL3 -DBUILD_SINGLE=1 -DBUILD_DOUBLE=1 -DBUILD_COMPLEX=1 -DBUILD_COMPLEX16=1 -DVERSION=\"0.3.28\" -UASMNAME -UASMFNAME -UNAME -UCNAME -UCHAR_NAME -UCHAR_CNAME -DASMNAME=spotrf_L_parallel -DASMFNAME=spotrf_L_parallel_ -DNAME=spotrf_L_parallel_ -DCNAME=spotrf_L_parallel -DCHAR_NAME=\"spotrf_L_parallel_\" -DCHAR_CNAME=\"spotrf_L_parallel\" -DNO_AFFINITY -I../.. -UDOUBLE  -UCOMPLEX -UCOMPLEX -UDOUBLE potrf_L_parallel.c -o spotrf_L_parallel.o
potrf_U_parallel.c: In function ‘spotrf_U_parallel’:
potrf_U_parallel.c:124:40: error: passing argument 5 of ‘syrk_thread’ from incompatible pointer type [-Wincompatible-pointer-types]
  124 |                   &newarg, NULL, NULL, (int (*)(void))HERK_UC, sa, sb, args -> nthreads);
In file included from ../../common.h:849,
                 from potrf_U_parallel.c:40:
../../common_thread.h:212:71: note: expected ‘int (*)(blas_arg_t *, BLASLONG *, BLASLONG *, float *, float *, BLASLONG)’ {aka ‘int (*)(blas_arg_t *, long int *, long int *, float *, float *, long int)’} but argument is of type ‘int (*)(void)’
  212 | int syrk_thread(int mode, blas_arg_t *, BLASLONG *, BLASLONG *, int (*function)(blas_arg_t*, BLASLONG*, BLASLONG*, FLOAT *, FLOAT *, BLASLONG), void*, void*, BLASLONG);
      |                                                                 ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
potrf_L_parallel.c: In function ‘spotrf_L_parallel’:
potrf_L_parallel.c:124:40: error: passing argument 5 of ‘syrk_thread’ from incompatible pointer type [-Wincompatible-pointer-types]
  124 |                   &newarg, NULL, NULL, (int (*)(void))HERK_LN, sa, sb, args -> nthreads);
In file included from ../../common.h:849,
                 from potrf_L_parallel.c:40:
../../common_thread.h:212:71: note: expected ‘int (*)(blas_arg_t *, BLASLONG *, BLASLONG *, float *, float *, BLASLONG)’ {aka ‘int (*)(blas_arg_t *, long int *, long int *, float *, float *, long int)’} but argument is of type ‘int (*)(void)’
  212 | int syrk_thread(int mode, blas_arg_t *, BLASLONG *, BLASLONG *, int (*function)(blas_arg_t*, BLASLONG*, BLASLONG*, FLOAT *, FLOAT *, BLASLONG), void*, void*, BLASLONG);
      |                                                                 ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Full log available here: https://buildd.debian.org/status/fetch.php?pkg=openblas&arch=mips64el&ver=0.3.28%2Bds-1&stamp=1723372492&raw=0

--- a/lapack/potrf/potrf_L_parallel.c
+++ b/lapack/potrf/potrf_L_parallel.c
@@ -121,7 +121,7 @@ blasint CNAME(blas_arg_t *args, BLASLONG *range_m, BLASLONG *range_n, FLOAT *sa,
       HERK_THREAD_LN(&newarg, NULL, NULL, sa, sb, 0);
 #else
       syrk_thread(mode | BLAS_TRANSA_N | BLAS_TRANSB_T | BLAS_UPLO,
-		  &newarg, NULL, NULL, (int (*)(void))HERK_LN, sa, sb, args -> nthreads);
+		  &newarg, NULL, NULL, (int (*)(blas_arg_t *, BLASLONG *, BLASLONG *, FLOAT *, FLOAT *, BLASLONG))HERK_LN, sa, sb, args -> nthreads);
 #endif
     }
   }

--- a/lapack/potrf/potrf_U_parallel.c
+++ b/lapack/potrf/potrf_U_parallel.c
@@ -121,7 +121,7 @@ blasint CNAME(blas_arg_t *args, BLASLONG *range_m, BLASLONG *range_n, FLOAT *sa,
       HERK_THREAD_UC(&newarg, NULL, NULL, sa, sb, 0);
 #else
       syrk_thread(mode | BLAS_TRANSA_N | BLAS_TRANSB_T,
-		  &newarg, NULL, NULL, (int (*)(void))HERK_UC, sa, sb, args -> nthreads);
+		  &newarg, NULL, NULL, (int (*)(blas_arg_t *, BLASLONG *, BLASLONG *, FLOAT *, FLOAT *, BLASLONG))HERK_UC, sa, sb, args -> nthreads);
 #endif
     }
   }

